rust/kcl-lib/std/vector.kcl

```diff
@@ -0,0 +1,87 @@
+/// Adds every element of v0 to its corresponding element in v1.
+/// Both vectors must have the same length. Returns a new vector of the same length.
+/// In other words, component-wise addition.
+/// ```kcl
+/// v0 = [1, 2, 3]
+/// v1 = [10, 10, 10]
+/// v2 = vector::add(v0, v1)
+/// assert(v2[0], isEqualTo = 11)
+/// assert(v2[1], isEqualTo = 12)
+/// assert(v2[2], isEqualTo = 13)
+/// ```
+export fn add(@v0: [number], v1: [number]): [number] {
+  l0 = reduce(v0, initial = 0, f = fn(@i, accum) { return accum + 1})
+  l1 = reduce(v1, initial = 0, f = fn(@i, accum) { return accum + 1})
```

Comment:
Should we pull this out to a helper function? Formatting this file ends up in a very unpleasant result. Eventually it should be exposed in the stdlib #4804, but presumably you could make an unexported function for now.